aio_pika/robust_connection.py

```diff
@@ -77,44 +91,61 @@ def add_reconnect_callback(self, callback: Callable[[], None]):
 
         self._reconnect_callbacks.add(callback)
 
+    async def __cleanup_connection(self, exc):
+        await asyncio.gather(
+            self.connection.close(exc),
+            return_exceptions=True,
+        )
+        self.connection = None
+
     async def connect(self, timeout: TimeoutType = None, **kwargs):
+        if self.is_closed:
+            raise RuntimeError("{!r} connection closed".format(self))
+
         if kwargs:
             # Store connect kwargs for reconnects
             self.connect_kwargs = kwargs
 
-        while True:
-            try:
-                return await super().connect(
-                    timeout=timeout, **self.connect_kwargs
-                )
-            except CONNECTION_EXCEPTIONS:
-                if self.fail_fast:
-                    raise
-
-                log.warning(
-                    "First connection attempt failed "
-                    "and will be retried after %d seconds",
-                    self.reconnect_interval,
-                    exc_info=True,
-                )
+        if self.reconnecting:
```

Comment:
Why not to return if already reconnecting?

Reply:
when the user wants to call reconnect explicitly it's just printed log warning.